src/viadot/orchestration/prefect/tasks/adls.py

```diff
@@ -90,3 +91,119 @@ def df_to_adls(
         sep=sep,
         overwrite=overwrite,
     )
+
+
+@task(retries=3, retry_delay_seconds=10)
+def adls_to_df(
+    path: str,
+    sep: str = "\t",
+    credentials_secret: str | None = None,
+    config_key: str | None = None,
+) -> pd.DataFrame:
+    r"""Load file(s) from the Azure Data Lake to a pandas DataFrame.
+
+    Note: Currently supports CSV and parquet files.
+
+    Args:
+        path (str): The path from which to load the DataFrame.
+        sep (str, optional): The separator to use when reading a CSV file.
+            Defaults to "\t".
+        credentials_secret (str, optional): The name of the Azure Key Vault secret
+            storing the credentials.
+        config_key (str, optional): The key in the viadot config holding relevant
+            credentials.
+
+    Raises:
+        MissingSourceCredentialsError: If credentials were not provided.
+
+    Returns:
+        pd.DataFrame: The HTTP response object.
+    """
+    logger = get_run_logger()
+
+    if not (credentials_secret or config_key):
+        raise MissingSourceCredentialsError
+
+    credentials = get_credentials(credentials_secret)
+    lake = AzureDataLake(credentials=credentials, config_key=config_key)
+
+    full_dl_path = str(Path(credentials["ACCOUNT_NAME"], path))
+    logger.info(f"Downloading data from {full_dl_path} to a DataFrame...")
+
+    name = Path(path).stem
+    suffixes = "".join(Path(path).suffixes)
+    file_name = f"{name}{suffixes}"
+
+    lake.download(to_path=file_name, from_path=path, recursive=False)
+
+    if ".csv" in suffixes:
+        df = pd.read_csv(file_name, sep=sep)
+    elif ".parquet" in suffixes:
+        df = pd.read_parquet(file_name)
+
+    Path.unlink(file_name)
+
+    logger.info("Successfully loaded data.")
+
+    return df
```

Comment:
There's already an `ADLS.to_df()` method than you can use for this task - no need to reinvent the wheel

Reply:
That is not working. I created it because Fabio asked me to.

Reply:
What's not working? Can you link the issue? Also, that means you should fix the issue and not add another function that does the same thing.